provider/provider.go

```diff
@@ -102,6 +102,34 @@ func (p *Provider) ConfigureProvider(ctx context.Context, request *cqproto.Confi
 	return &cqproto.ConfigureProviderResponse{}, nil
 }
 
+func (p *Provider) CheckDuplicates() error {
```

Comment:
This should run as part of the Configure function that function is basically the provider's "initialize" function